packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/templates/defaultSearchBoxInput.html

```diff
@@ -37,6 +37,8 @@
               <template if:false={isQueryEmpty}>
                 <lightning-button-icon
                   onclick={clearInput}
+                  onkeyup={onKeyup}
```

Comment:
this supports the case where a key is pressed and triggers the `handleEnter()` function which will see that the clear icon is indeed focused and therefore call `clearInput()`

Reply:
How was the bug happening before the existence of this listener? how and why  was clicking enter on this icon triggering a search?

Reply:
1- For your question about whether it was a standalone searchbox only issue: no it was a behaviour that was happening in both searchboxes (where tab and then pressing ENTER on clear icon would clear + trigger a search with empty query) whereas we want : to simply clear the input without triggering a search.

2- This listener was indeed useless. I thought I needed it because it was triggering the `onkeyup` listener of the input. What I found was that when we were calling `clearInput()` we were setting focus on the input and therefore I believe the browser was clearing and then triggering a `keyup` event on the input which had focus at that time. 

I removed the listener and remove the focus on input while clearing the input. 

3- BONUS: I noticed that if I tab to focus on clearIcon and then shift+tab to focus back on the input and then pressed ENTER, it would no longer trigger a search. This is why I added the `onblur()` listener to remove focus from the `clearIcon`